keeparelease/keeparelease.go

```diff
@@ -0,0 +1,72 @@
+package keeparelease
+
+import (
+	"errors"
+	"io/ioutil"
+	"regexp"
+	"strings"
+)
+
+const regexTitle string = `^\#\#\s+\[{1,2}(?P<full_version>(?P<major>(?:0|[1-9][0-9]*))\.(?P<minor>(?:0|[1-9][0-9]*))\.(?P<patch>(?:0|[1-9][0-9]*))(\-(?P<prerelease>(?:0|[1-9A-Za-z-][0-9A-Za-z-]*)(\.(?:0|[1-9A-Za-z-][0-9A-Za-z-]*))*))?(\+(?P<build>[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?)\]{1,2}\s+-?\s+(?P<date>.*)`
+
+// ParseChangelog parses a ChangeLog respecting the Keep A Changelog format.
+func ParseChangelog(changelog []string) (string, string, error) {
+	re := regexp.MustCompile(regexTitle)
+	title := ""
+	releaseInfo := make([]string, 1)
```

Comment:
This looks like a bug -- this makes a slice of length one with one empty string in it. It's basically the equivalent of:

```go
releaseInfo := []string{""}
```

This means that your check below for `if len(releaseInfo) == 0` will always be false.

Is there any reason you want to initialize `releaseInfo` to start with anything in it? I'd think you'd be totally fine replacing this line with `var releaseInfo []string`.